driver/fuzz_target_runner.cpp

```diff
@@ -12,75 +12,179 @@
-              "--keep_going=N to stop after N findings.");
-DEFINE_string(autofuzz_ignore, "",
-              "Fully qualified class names of exceptions to ignore during "
-              "autofuzz. Separated by comma.");
-DEFINE_bool(
-    fake_pcs, false,
-    "Supply synthetic Java program counters to libFuzzer trace hooks to "
-    "make value profiling more effective. Enabled by default if "
-    "-use_value_profile=1 is specified.");
-
-DECLARE_bool(hooks);
+
+#include "com_code_intelligence_jazzer_driver_FuzzTargetRunner.h"
+#include "driver/fuzzed_data_provider.h"
+
+extern "C" int LLVMFuzzerRunDriver(int *argc, char ***argv,
+                                   int (*UserCb)(const uint8_t *Data,
+                                                 size_t Size));
+
+namespace {
+bool gUseFuzzedDataProvider;
+jclass gRunner;
+jmethodID gRunOneId;
+JNIEnv *gEnv;
+
+// A libFuzzer-registered callback that outputs the crashing input, but does
+// not include a stack trace.
+void (*gLibfuzzerPrintCrashingInput)() = nullptr;
+
+int testOneInput(const uint8_t *data, const std::size_t size) {
+  JNIEnv &env = *gEnv;
+  jbyteArray input = nullptr;
+  jint jsize =
+      std::min(size, static_cast<size_t>(std::numeric_limits<jint>::max()));
+  if (gUseFuzzedDataProvider) {
+    ::jazzer::FeedFuzzedDataProvider(data, size);
+  } else {
+    input = env.NewByteArray(jsize);
+    env.SetByteArrayRegion(input, 0, jsize,
+                           reinterpret_cast<const jbyte *>(data));
+  }
+  int res = env.CallStaticIntMethod(gRunner, gRunOneId, input);
+  if (env.ExceptionCheck()) {
+    env.ExceptionDescribe();
+    _Exit(1);
+  }
+  return res;
+}
+}  // namespace
 
 namespace jazzer {
-std::vector<std::string> fuzzTargetRunnerFlagsAsDefines() {
-  return {
-      absl::StrFormat("-Djazzer.target_class=%s", FLAGS_target_class),
-      absl::StrFormat("-Djazzer.target_args=%s", FLAGS_target_args),
-      absl::StrFormat("-Djazzer.keep_going=%d", FLAGS_keep_going),
-      absl::StrFormat("-Djazzer.dedup=%s", FLAGS_dedup ? "true" : "false"),
-      absl::StrFormat("-Djazzer.ignore=%s", FLAGS_ignore),
-      absl::StrFormat("-Djazzer.reproducer_path=%s", FLAGS_reproducer_path),
-      absl::StrFormat("-Djazzer.coverage_report=%s", FLAGS_coverage_report),
-      absl::StrFormat("-Djazzer.coverage_dump=%s", FLAGS_coverage_dump),
-      absl::StrFormat("-Djazzer.autofuzz=%s", FLAGS_autofuzz),
-      absl::StrFormat("-Djazzer.autofuzz_ignore=%s", FLAGS_autofuzz_ignore),
-      absl::StrFormat("-Djazzer.hooks=%s", FLAGS_hooks ? "true" : "false"),
-      absl::StrFormat("-Djazzer.fake_pcs=%s",
-                      FLAGS_fake_pcs ? "true" : "false"),
-  };
+void DumpJvmStackTraces() {
+  JavaVM *vm;
+  jsize num_vms;
+  JNI_GetCreatedJavaVMs(&vm, 1, &num_vms);
+  if (num_vms != 1) {
+    return;
+  }
+  JNIEnv *env = nullptr;
+  if (vm->AttachCurrentThread(reinterpret_cast<void **>(&env), nullptr) !=
+      JNI_OK) {
```

Comment:
Are there cases where this is expected to fail? Do you think it makes sense log a warning here?

Reply:
I don't know, I added a warning to be on the safe side.